docs/SPEC.md

```diff
@@ -3685,31 +3685,39 @@ from(bucket: "telegraf/autogen")
-| fieldFn    | (r: record) -> record | Function that takes a record from the input table and returns an object. For each record from the input table `fieldFn` returns on object that maps output field key to output value. Default: `(r) => ({ [r._field]: r._value })` |
-
-TODO(nathanielc): The fieldFn is not valid and needs to change. It uses dynamic object keys which is not allowed.
+The `to` operation takes data from a stream and writes it to a bucket.
+`to` has the following properties:
+
+| Name              | Type           | Description                                                                                                                                                                                                                        |
+| ----              | ----           | -----------                                                                                                                                                                                                                        |
+| bucket            | string         | Bucket is the bucket name into which data will be written.                                                                                                                                                                         |
+| bucketID          | string         | BucketID is the bucket ID into which data will be written.                                                                                                                                                                         |
+| org               | string         | Org is the organization name of the bucket.                                                                                                                                                                                        |
+| orgID             | string         | OrgID is the organization ID of the bucket.                                                                                                                                                                                        |
+| host              | string         | Host is the location of a remote host to write to. Defaults to `""`.                                                                                                                                                               |
+| token             | string         | Token is the authorization token to use when writing to a remote host. Defaults to `""`.                                                                                                                                           |
+| measurementColumn | string         | MeasurementColumn is the measurement into which data will be written. Defaults to `"_measurement"`.                                                                                                                                |
```

Comment:
I would like to see these parameters removed (measurementColumn, fieldColumn, etc.) and instead have the input schema be enforced by the type system (eventually). In other words a user must provide `to` with tables having `_time`, `_field`, `_value`, `_start`, and `_stop` columns. The rest of the column types being string.

Reply:
@jlapacik is this because it simplifies the type system to have the column names static?

If so what about the pattern of most other Flux functions to take a `column` parameter? 

Does the `label type` (aka const string type) solve this problem?

Reply:
@nathanielc yes as it stands right now (that is, without label types) in order to assign a valid type signature to `to`, the names and types of the columns must be known at compile time. This precludes us from passing the column names as input to the function itself.

I _think_ we could get around this with label types, but that would come with the added complexity of an AST preprocessor (or something similar). Although we'll probably need that anyway since like you mention, functions like `keep` and `drop` have the same issue.

Reply:
i'm not convinced this is the right move.  we set those as defaults to get this behavior, and I don't want to see the type system becoming an impediment to writing data to influxdb.  Furthermore, if we were being smart with our planner, any renaming should be rolled into this function anyway.  This idea also falls under the category of being "too helpful"  where the easy path feels great but doing anything advanced or unorthodox requires a herculean effort.  

Also, currently, a measurement name is required for writing data, I don't think we can work around that at this time.  One of the tags has to be specially earmarked as the measurement name.  

I would, however, prefer to eliminate the _field column.  But that's a different discussion -- I want it to be eliminated from the top of the chain to the bottom.  Field names should be column labels, period, and things like a default _value should be used more sparingly.  In theory, this would be a good place to start but it's awkward to require writing data back to influxdb in a format that's different than how it comes out.